nav2_tasks/include/nav2_tasks/rate_controller_node.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+  
+#ifndef NAV2_TASKS__RATE_CONTROLLER_NODE_HPP_
+#define NAV2_TASKS__RATE_CONTROLLER_NODE_HPP_
+
+#include <chrono>
+#include "behavior_tree_core/decorator_node.h"
+
+namespace nav2_tasks
+{
+
+class RateController : public BT::DecoratorNode
```

Comment:
This would be a great addition to the BT library.

Reply:
Yes, Davide was interested in this and I pointed him to my code (https://github.com/BehaviorTree/BehaviorTree.CPP/issues/13). I'm sure he'll implement something.